### PR TITLE
Adding unit tests for TokenMgtDAO

### PR DESCRIPTION
**Pull request by thanujalk**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 1]
```diff
@@ -141,21 +142,37 @@ public void initTest() throws Exception {
...
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME
+
+                },
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        SAMPLE_DOMAIN
+
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: extra new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 2]
```diff
@@ -141,21 +142,37 @@ public void initTest() throws Exception {
                 {
                         CALLBACK_URL,
                         MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
-                        MultitenantConstants.SUPER_TENANT_ID
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME
+
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: extra new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java (hunk 1 above)
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java (hunk 2 above)
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 3]
```diff
@@ -165,21 +182,38 @@ public void testStoreAuthorizationCode(String callbackUrl, String tenantDomain,
...
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: extra new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 4]
```diff
@@ -165,21 +182,38 @@ public void testStoreAuthorizationCode(String callbackUrl, String tenantDomain,
...
+                },
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        SAMPLE_DOMAIN
+
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: extra new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 5]
```diff
@@ -141,21 +142,37 @@ public void initTest() throws Exception {
...
+                        SAMPLE_TENANT_ID,
+                        SAMPLE_DOMAIN
                 },
         };
     }
 
     @Test(dataProvider = "storeAuthorizationCodeDataProvider")
-    public void testStoreAuthorizationCode(String callbackUrl, String tenantDomain, int tenantId) throws
-            IdentityOAuth2Exception, SQLException, IdentityOAuthAdminException {
+    public void testStoreAuthorizationCode(String callbackUrl, String tenantDomain, int tenantId, String
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Just throw generic Exception.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 6]
```diff
@@ -189,23 +223,38 @@ public void testPersistAuthorizationCode(String callbackUrl, String tenantDomain
                 {
                         CALLBACK_URL,
                         MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
-                        MultitenantConstants.SUPER_TENANT_ID
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME
+
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: extra new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java (hunk 3 above)
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java (hunk 4 above)
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java (hunk 6 above)
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 7]
```diff
@@ -189,23 +223,38 @@ public void testPersistAuthorizationCode(String callbackUrl, String tenantDomain
...
+                },
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        SAMPLE_DOMAIN
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 8]
```diff
@@ -189,23 +223,38 @@ public void testPersistAuthorizationCode(String callbackUrl, String tenantDomain
...
                 },
         };
     }
 
     @Test(dataProvider = "deactivateAuthorizationCodeDataProvider")
-    public void testDeactivateAuthorizationCode(String callbackUrl, String tenantDomain, int tenantId) throws
-            SQLException, IdentityOAuth2Exception {
-        createApplication(UUID.randomUUID().toString(), UUID.randomUUID().toString(), tenantId);
+    public void testDeactivateAuthorizationCode(String callbackUrl, String tenantDomain, int tenantId,
+                                                String userStoreDomain) throws SQLException, IdentityOAuth2Exception {
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: throw Exception

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 9]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME,
+                        OAuthConstants.UserType.APPLICATION,
+                        OAuthConstants.GrantTypes.AUTHORIZATION_CODE,
+                        false
+                },
+        };
+    }
+
+    @Test(dataProvider = "retrieveAccessTokensDataProvider")
+    public void testRetrieveAccessTokens(String tenantDomain, int tenantId, String userStoreDomain,
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Throw Exception

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 10]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
-        authenticatedUser.setTenantDomain(tenantDomain);
-        authenticatedUser.setUserStoreDomain(UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+        String consumerKey = UUID.randomUUID().toString();
+        createApplication(consumerKey, UUID.randomUUID().toString(), tenantId);
+        AuthenticatedUser authenticatedUser = getAuthenticatedUser(tenantDomain, userStoreDomain);
+
+        AccessTokenDO expiredAccessToken = getAccessTokenDO(consumerKey, authenticatedUser, applicationType,
+                tenantId, grantType);
+        expiredAccessToken.setTokenState(OAuthConstants.TokenStates.TOKEN_STATE_EXPIRED);
+        try (Connection connection = DAOUtils.getConnection(DB_NAME)) {
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Any reason for having a try block ?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 11]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+    @DataProvider(name = "validateAuthorizationCodeDataProvider")
+    public Object[][] validateAuthorizationCodeData() {
+
+        return new Object[][]{
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 12]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+                },
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        SAMPLE_DOMAIN
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 13]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+    public Object[][] changeAuthzCodeStateData() {
+
+        return new Object[][]{
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME,
+                        OAuthConstants.AuthorizationCodeState.EXPIRED
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java (hunk 10 above)
Comment (thanujalk) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: close the connection

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 14]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+                },
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        SAMPLE_DOMAIN,
+                        OAuthConstants.AuthorizationCodeState.EXPIRED
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 15]
```diff
@@ -532,50 +581,818 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+    @DataProvider(name = "deactivateAuthorizationCodesDataProvider")
+    public Object[][] deactivateAuthorizationCodesData() {
+
+        return new Object[][]{
+                {
+                        CALLBACK_URL,
+                        MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                        MultitenantConstants.SUPER_TENANT_ID,
+                        UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: Remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 16]
```diff
@@ -532,53 +572,804 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+        try (Connection connection = DAOUtils.getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            tokenMgtDAO.setAccessTokenState(connection, accessTokenDO.getTokenId(), tokenState, UUID.randomUUID()
+                    .toString(), userStoreDomain);
+            assertTrue(tokenState.equals(getAccessTokenStatusByTokenId(accessTokenDO.getTokenId())), "Failed to set " +
+                    "access token status.");
+        }
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: remove unnecessary new line

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java [hunk 17]
```diff
@@ -532,53 +572,804 @@ public void testRetrieveLatestAccessToken(String tenantDomain, int tenantId, Str
...
+        AuthzCodeDO authzCodeDO = new AuthzCodeDO(authenticatedUser, new String[]{"scope1", "scope2"},
+                new Timestamp(System.currentTimeMillis()), 3600L, callbackUrl, consumerKey, authzCode, authzCodeId,
+                null, null);
+        try (Connection connection = DAOUtils.getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            tokenMgtDAO.storeAuthorizationCode(authzCode, consumerKey, callbackUrl, authzCodeDO);
+            assertNotNull(tokenMgtDAO.getAuthzCodeByCodeId(authzCodeId), "Failed to persist authorize code.");
+        }
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAOTest.java]: remove unnecessary new line

**Pull request merged**